src/utils/getFirstAndLastSceneIndicesForInterpolationInputRange.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+/* eslint-disable */
+
+import type {
+  NavigationSceneRendererProps,
+  NavigationScene,
+  SceneIndicesForInterpolationInputRange,
+} from '../TypeDefinition';
+
+function getFirstAndLastSceneIndicesForInterpolationInputRange(
+  props: NavigationSceneRendererProps
+): SceneIndicesForInterpolationInputRange | null {
+  const { scene, scenes } = props;
+  const index = scene.index;
+  const lastSceneIndicesInQueue = scenes.length - 1
+  const isBack = !scenes[lastSceneIndicesInQueue].isActive
+
+  if (isBack) {
+    const currentSceneIndicesInQueue = scenes.findIndex((item: NavigationScene) => item === scene);
```

Comment:
hi! Thanks for being so quick! I'm really sorry, but the naming still has room for improvement. My main point was to remove possible confusion between *active* and *current*. Now we have `currentSceneIndicesInQueue` and `activeSceneIndicesInQueue`. The difference between these is still hard to tell. Moreover, you used 'indices', which is a plural and that suggests the variable is some sort of collection of values, but it is only a single number. Also, I'm not quite sure about the word 'queue' since that gives an impression of some FIFO stuff going on, and I thinks it's better to think of `scenes` as of a list/array (js doesn't really make a difference between these two). I'm guessing English is not your first language, so if you feel like these differences are beyond your resolving power I can rename the variables, push the changes to your branch and then merge. It's still going to count as your contribution. This is just an offer, you can give it another shot if you want :)

Reply:
Yes, Chinese is my mother tongue. But here is indeed confusing, because a scene with two index, one is called the key of the index, the other is in the scene stack index. This time I re-named, I do not know whether to meet your naming norms, if still not through, trouble you help me to rename it, thank you.